### DISCUSSION TITLE
Retaining conversational elements

### DISCUSSION THREAD

**Discussion by madscientistca** [Future-House/paper-qa#105] [General]
It does not appear that how the query function works that it retains any conversational memory as you query a group of papers. It seems that when I am querying through an interface i built, from question to question there is no retention as I am accustom to using the chatgpt web interface.  

Is there an "easy" way to hack the docs file to accommodate a conversational component similar to what the completion API does?  I guess I could just string the history of queries and responses together as part of each subsequent query?  Although the issue there then is that unlike the completion API, there are no role assignments and likely the number of tokens would quickly exceed the capacity of the query interface?  

Only raising because that ability to contextualize answers to current queries with respect to previous queries and responses, definitely leads to superior answers.

**Comment by stonebyun** [comment]
This might be an exceedingly vague and abstract expression of what I understood of your question. 
Your matter seems like the same as the matter of understanding how human mind keeps his/her oneness of consciousness or identity in time and space.

**Comment by whitead** [comment]
You can now add memory - `docs(memory=True)` - so that it remembers previous answers
